python/graphstorm/model/gnn_with_reconstruct.py

```diff
@@ -14,13 +14,94 @@
     limitations under the License.
 """
 
+from functools import partial
+
 import torch as th
 import dgl
-from dgl.distributed import DistTensor, node_split
+from dgl.distributed import node_split
 
 from ..utils import barrier
 from .gnn_encoder_base import GraphConvEncoder, dist_inference_one_layer
 
+def construct_node_feat(g, rel_names, input_gnn, get_input_embeds, batch_size,
+                        edge_mask=None, device="cpu", task_tracker=None):
+    """ Construct node features with the input layer.
+
+    Parameters
+    ----------
+    g : DistGraph
+        The distributed graph.
+    rel_names : list
+        The relation types used to construct node features.
+    input_encoder : GraphConvEncoder
+        The encoder to construct node features.
+    get_input_embeds : callable
+        Get the node features of the input nodes.
+    batch_size : int
+        The batch size for the GNN inference.
+    edge_mask : str
+        The edge mask indicates which edges are used to compute GNN embeddings.
+    device : Pytorch Device
+        The device where to perform the computation.
+    task_tracker : GSTaskTrackerAbc
+        The task tracker.
+
+    Returns
+    -------
+    dict of Tensor : the constructed node features.
+    """
+    # Here we only need to compute embeddings for the destination node types
+    # of the required relation types.
+    target_ntypes = {rel_name[2] for rel_name in rel_names}
+    with th.no_grad():
+        infer_nodes = {}
+        for ntype in target_ntypes:
+            infer_nodes[ntype] = node_split(th.ones((g.number_of_nodes(ntype),),
+                                                    dtype=th.bool),
+                                            partition_book=g.get_partition_book(),
+                                            ntype=ntype, force_even=False)
+        # We use all neighbors to reconstruct node features.
+        sampler = dgl.dataloading.MultiLayerNeighborSampler([-1], mask=edge_mask)
```

Comment:
Why not use construct_node_feat?

Reply:
i guess you mean why not using `construct_node_fanout`. It's a bit hard to pass the argument. this function is used for full-graph inference.